profondo/cnn.py

```diff
@@ -126,21 +226,21 @@ def report_callback(epoch, model, x_val, y_val):
     meta, images = load_data(args.input_dir)
     x_train, y_train, x_test, y_test, x_val, y_val = train_test_split(meta, images)
 
-    # See sklearn.utils.class_weight.compute_class_weight
-    genre_weights = y_train.sum()/(y_train.shape[1]*y_train.sum(axis=0).astype(float))
-    genre_weights = dict(enumerate(genre_weights))
+    genre_weights = compute_genre_weights(y_train)
     logging.info("Genre weights: {}".format(genre_weights))
 
     callbacks = [
         ModelCheckpoint(filepath=MODEL_FILENAME, monitor='val_loss', save_best_only=True, verbose=1),
         EarlyStopping(monitor='val_loss', min_delta=0, patience=10, verbose=1, mode='auto'),
+        TensorBoard(log_dir=join(TB_DIR, 'full'), histogram_freq=1, write_graph=True, write_images=False),
         LambdaCallback(on_epoch_end=lambda e, l: report_callback(e, model, x_val, y_val))
     ]
 
     train_datagen = ImageDataGenerator()
     train_generator = train_datagen.flow(x_train, y_train, batch_size=args.batch_size)
 
-    model = load_model(x_train, y_train)
+    model = build_pre_trained(args.pre_trained, x_train, y_train, x_val, y_val)\
```

Comment:
We should use the non-pretrained model if no pre-trained network name is specified.